server/src/main/java/io/crate/metadata/SimpleReference.java

```diff
@@ -262,6 +262,10 @@ public Map<String, Object> toMapping(int position, @Nullable Metadata.ColumnOidS
         mapping.put("position", position);
         if (oid == COLUMN_OID_UNASSIGNED && columnOidSupplier != null) {
             mapping.put("oid", columnOidSupplier.nextOid());
+        } else if (oid != COLUMN_OID_UNASSIGNED) {
+            // Print out existing oid without shifting a global one.
+            // Mainly needed for tests, so that printed out mapping actually corresponds to the real mapping (see ReferenceTest).
```

Comment:
This comment is confusing. We always must write any assigned OID back to the cluster state, this is not a print out, nor is this only related to tests. If we wouldn't we may loose OID's when modifying a table (and such writing it's metadata).

Reply:
We shouldn't lose OID if it's assigned since mapping["oid"] is not used for persisting metadata. Usually we use mapping fields but oid is an exception since we have dedicated Metatadata.Builder.Supplier object which we propagate further down to ES components to be safe.
related: https://github.com/crate/crate/pull/14142#discussion_r1199985838

This change was done since `ReferenceTest` was not able to show OID (even if Reference had OID) ie. `Reference.toMapping` could only show "fresh, just assigned OIDS" but not existing ones.

Reply:
maybe we need a javadoc that `toMapping` doesn't affect OID in a cluster state for existing references?

Reply:
I see, I will drop the comment then -  we can treat this an improvement?

Reply:
👍 thanks :)